### PR TITLE
docs: add supported features to payment service definition

### PR DESCRIPTION
**Pull request by douglaseggleton**
# Description

https://sprints.zoho.com/team/gr4vy#itemdetails/P10/I256

Screenshot 2021-08-03 at 11 54 20

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own changes
- [ ] I have run `yarn test` to make sure my changes pass all linters
- [ ] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: reference/resources/payment_service_definition.yaml [hunk 1]
```diff
@@ -95,6 +95,24 @@ properties:
...
+  supported_features:
+    type: object
+    description: |-
+      Features supported by the payment definition
+    properties:
+      three_d_secure_pass_through: 
+        type: boolean
+        example: false
+        description: Supports passing 3D secure data to the underlying processor
+      payment_method_tokenization:
```
Comment (ipeluffo) [inline reference/resources/payment_service_definition.yaml:107]:
```suggestion
      payment_tokenization:
```

File: reference/resources/payment_service_definition.yaml [hunk 2]
```diff
@@ -95,6 +95,24 @@ properties:
...
+        type: boolean
+        example: true
+        description: Supports storing a payment method via tokenization
+      three_d_secure:
```
Comment (ipeluffo) [inline reference/resources/payment_service_definition.yaml:111]:
```suggestion
      three_d_secure_hosted:
```

File: reference/resources/payment_service_definition.yaml [hunk 3]
```diff
@@ -95,6 +95,24 @@ properties:
       - GB
       - DE
 
+  supported_features:
```
Comment (ipeluffo) [inline reference/resources/payment_service_definition.yaml:98]: Should we order them alphabetically?

**Review by ipeluffo — COMMENTED** [review]
There are a few linting issues that should be fixed. Thanks for opening this PR :)

File: reference/resources/payment_service_definition.yaml [hunk 4]
```diff
@@ -95,6 +95,24 @@ properties:
...
+    type: object
+    description: |-
+      Features supported by the payment definition.
+    properties:
+      payment_tokenization:
```
Comment (cbetta) [inline reference/resources/payment_service_definition.yaml:103]:
```suggestion
      payment_method_tokenization:
```

It's not the payment we're tokenizing is it?

File: reference/resources/payment_service_definition.yaml [hunk 5]
```diff
@@ -95,6 +95,24 @@ properties:
...
+        type: boolean
+        example: true
+        description: Supports storing a payment method via tokenization.
+      three_d_secure_hosted:
+        type: boolean
+        example: true
+        description: Supports hosted 3D secure with a redirect.
```
Comment (cbetta) [inline reference/resources/payment_service_definition.yaml:110]:
```suggestion
        description: Supports hosted 3-D Secure with a redirect.
```

File: reference/resources/payment_service_definition.yaml [hunk 6]
```diff
@@ -95,6 +95,24 @@ properties:
...
+      three_d_secure_pass_through: 
+        type: boolean
+        example: false
+        description: Supports passing 3D secure data to the underlying processor.
```
Comment (cbetta) [inline reference/resources/payment_service_definition.yaml:114]:
```suggestion
        description: Supports passing 3-D Secure data to the underlying processor.
```

I think this is the official spelling.

File: reference/resources/payment_service_definition.yaml (hunk 4 above)
Comment (douglaseggleton) [reply reference/resources/payment_service_definition.yaml:103]: @ipeluffo Does this need to change in the payment definition?

Comment (ipeluffo) [reply reference/resources/payment_service_definition.yaml:103]: @douglaseggleton , I'll apply this change in `core-api` PR to match @cbetta 's suggestion

**Review by ipeluffo — APPROVED** [review]
gif

**Pull request merged**